### ISSUE TITLE
Blank screen when "Remote JS Debugging" is disabled

### ISSUE THREAD

**Issue by gtRfnkN** [microsoft/react-native-windows#6489]
We've upgraded our app from RNW 0.59 to 0.63. All js and native modules are migrated and everything works just fine with "Remote JS Debugging" turned on. When turned off, the app just shows a blank white screen without any useful warnings or errors. The only output that we get is from the react-native server:
```
ERROR    TypeError: Unable to get property 'indexOf' of undefined or null reference        
ERROR    Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
LOG      Unable to symbolicate stack trace: The stack is null
ERROR    Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
LOG      Unable to symbolicate stack trace: The stack is null
LOG      Unable to symbolicate stack trace: The stack is null
```

Due to a lack of any warning & errors, we tried removing components to find out which one might cause the error. Having a single view with a bg color works. When addding back components and modules one-by-one, some work just fine but other cause the "same" error (based on the very limited output). We have also tried creating a new, blank rnw project and using only our js code, but that leads to the same issue.

Do you have any idea on what the cause for this could be? Or how we can debug it other than just removing/adding components one-by-one?

**Comment by asklar** [comment]
You should be getting a RedBox; maybe there is a period of time during which JS errors aren't being propagated to the redbox.
Are you able to share a repro project that shows this?

**Comment by gtRfnkN** [comment]
The RedBox does show up when an update comes in via Fast Refresh, but the message there also only states the `indexOf` error without any useful stack trace. This seems to be due to the source maps not being built / delivered properly. I guess fixing that would be a big step forward. We can't share the project, but we did try out the calculator sample from the [React Native Samples and Templates repo](https://github.com/microsoft/react-native-windows-samples/tree/master/samples/Calculator) and don't get any source maps there as well...

In the meantime, we also created a new react-native project from a typescript template, installed rnw into that, and moved our code in there step-by-step. Now, everything does work but since there was a lot of refactoring, we can't pinpoint the intial issue.

So even though is doesn't block us anymore, it would be nice to find out the underlying problem.
